### DISCUSSION TITLE
Choosing the time when writing data

### DISCUSSION THREAD

**Discussion by christianaz** [FrandsenGroup/beams#61] [Ideas]
Hello! I have a quick question. So I'm trying to use the "Write" function to get .asy files for the data. The problem is that I only want data from the first 0.5 us of the run, but there doesn't seem to be an option to only get a part of the run. I see that I can adjust the bin size, but not the time. Is there an option that I am missing?

**Comment by benfrandsen** [comment]
I suppose we could add in this option, but on the other hand, users can always just choose to mask out or remove any of the data they don't want. For example, Christiana, if you only want to plot the data up to 0.5 us, you could do the following in python:

mask = (t < 0.5)
a_short = a[mask]
t_short = t[mask]

plt.plot(t_short, a_short)

Personally, I would prefer to get the whole data array and then just select the parts I want. But if this is annoying to users, it should be a pretty simple matter to have the user input starting and ending times for writing the file.

**Comment by aPeter1** [comment]
Generally I think it would be better to have the whole array so you don't need to write it out again if you end up needing other sections later, but it would definitely be a simple feature to add if you guys think it would be useful.

**Comment by christianaz** [comment]
Oh, I didn't think of doing it that way. That will probably work.
